homeassistant/components/kostal_plenticore/switch.py

```diff
@@ -0,0 +1,173 @@
+
+
+class PlenticoreDataSwitch(CoordinatorEntity, SwitchEntity):
+    def __init__(
+            self,
+            coordinator,
+            entry_id: str,
+            platform_name: str,
+            module_id: str,
+            data_id: str,
+            switch_name: str,
+            switch_data: dict[str, Any],
+            formatter: Callable[[str], Any],
+            device_info: DeviceInfo,
+    ):
+        """Create a new switch Entity for Plenticore process data."""
+        super().__init__(coordinator)
+        self.entry_id = entry_id
+        self.platform_name = platform_name
+        self.module_id = module_id
+        self.data_id = data_id
+        self._state: bool | None = None
+        self._last_run_success: bool | None = None
+        self._switch_name = switch_name
+        self._switch_data = switch_data
+        self._formatter = formatter
+
+        self._device_info = device_info
+
+    @property
+    def available(self) -> bool:
+        """Return if entity is available."""
+        return (
+                super().available
+                and self.coordinator.data is not None
+                and self.module_id in self.coordinator.data
+                and self.data_id in self.coordinator.data[self.module_id]
+        )
+
+    async def async_added_to_hass(self) -> None:
+        """Register this entity on the Update Coordinator."""
+        await super().async_added_to_hass()
+        self.coordinator.start_fetch_data(self.module_id, self.data_id)
+
+    async def async_will_remove_from_hass(self) -> None:
+        """Unregister this entity from the Update Coordinator."""
+        self.coordinator.stop_fetch_data(self.module_id, self.data_id)
+        await super().async_will_remove_from_hass()
+
+    async def async_turn_on(self) -> None:
+        """Turn device on."""
+        if await self.coordinator._async_write_data(self.module_id, {self.data_id: '1'}):
+            self._state = True
+            self._last_run_success = True
+            await self.coordinator.async_request_refresh()
+        else:
+            self._last_run_success = False
+
+    async def async_turn_off(self) -> None:
+        """Turn device off."""
+        if await self.coordinator._async_write_data(self.module_id, {self.data_id: '0'}):
+            self._state = False
+            self._last_run_success = True
+            await self.coordinator.async_request_refresh()
+        else:
+            self._last_run_success = False
+
+    @property
+    def device_info(self) -> DeviceInfo:
+        """Return the device info."""
+        return self._device_info
+
+    @property
+    def assumed_state(self) -> bool:
+        """Return true if unable to access real state of entity."""
+        if self.coordinator.data is None:
+            raw_value = True
+            _LOGGER.debug("NO assumed_state ")
+        else:
+            raw_value = int(self.coordinator.data[self.module_id][self.data_id])
+
+        return bool(raw_value)
+
```

Comment:
How is this helpful to the user?
Is it better to just mark the switch as unavailable?

Reply:
removed, was copied from an switchbot integration, does not make sense or does deliver valuable information